### ISSUE TITLE
Fix S1854 FP: Support interpolated raw string literals

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6301]
### Description

FP for S1854 (DeadStores).

Using C#11 interpolated raw string literals the rule is raising an issue.

### Repro steps

```csharp
string empty = "";
    string x = $"""{empty}"""; // FP (string is still empty, should be compliant)
    x = $"""{empty}Test""";
    Foo(x);

    string emptyMultiline = """

        """;
    string q = $"""{emptyMultiline}"""; // FP (string is still empty, should be compliant)
    q = $"""{emptyMultiline}Test""";
    Foo(q);
```

### Expected behavior

The rule should not raise an issue.

### Actual behavior

The rule does raise an issue.

**Comment by csaba-sagi-sonarsource** [comment]
The rule is still going to raise for these values as we intentionally don't want to evaluate the string interpolations to look for empty strings.

**Comment by andrei-epure-sonarsource** [comment]
the empty multi line isn't really empty, is it? doesn't it contain a newline character?

**Comment by csaba-sagi-sonarsource** [comment]
> the empty multi line isn't really empty, is it? doesn't it contain a newline character?
> 
> sharplab: ![image](https://user-images.githubusercontent.com/38876598/201029725-3c5d5595-f650-40dc-af6e-87376a9b3676.png)

For raw string literals:
 The newlines following the opening quote and preceding the closing quote aren't included in the final content.
 Any whitespace to the left of the closing double quotes will be removed from the string literal.
